src/component.py

```diff
@@ -2,162 +2,255 @@
 import logging
 import os
 import shutil
-from dataclasses import dataclass, field
+from csv import DictReader
 from pathlib import Path
 
-from keboola.component import ComponentBase, UserException
-from keboola.component.dao import TableDefinition, TableMetadata, SupportedDataTypes
 import duckdb
 from duckdb.duckdb import DuckDBPyConnection
+from keboola.component import ComponentBase, UserException
+from keboola.component.dao import TableDefinition, TableMetadata, SupportedDataTypes
+import fnmatch
+from typing import Union
 
 KEY_MODE = "mode"
-KEY_IN_TABLES = "in_tables"
+KEY_IN_TABLES = "input"
 KEY_QUERIES = "queries"
-KEY_OUT_TABLES = "out_tables"
+KEY_OUT_TABLES = "output"
+KEY_DETECT_TYPES = "detect_types"
 DUCK_DB_DIR = os.path.join(os.environ.get('TMPDIR', '/tmp'), 'duckdb')
 
 KEY_DEBUG = 'debug'
 MANDATORY_PARS = []
 
 
-@dataclass
-class TableDef:
-    path: str
-    file_name: str
-    is_sliced: bool
-    manifest: dict = field(default_factory=dict)
-
-
 class Component(ComponentBase):
 
     def __init__(self):
         ComponentBase.__init__(self)
         self.validate_configuration_parameters(MANDATORY_PARS)
         self._config = self.configuration.parameters
-        # initialize instance parameters
+        self._connection = self.init_connection()
+        self._in_tables = self.get_input_tables_definitions()
 
     def run(self):
-        self.cleanup_duckdb()
 
-        if self._config[KEY_MODE] == 'advanced':
+        if self._config.get(KEY_MODE) == 'advanced':
             self.advanced_mode()
         else:
             self.simple_mode()
 
-        self.cleanup_duckdb()
+    def init_connection(self) -> DuckDBPyConnection:
+        """
+                Returns connection to temporary DuckDB database
+                """
+        os.makedirs(DUCK_DB_DIR, exist_ok=True)
+        # TODO: On GCP consider changin tmp to /opt/tmp
+        config = dict(temp_directory='/opt/tmp/dbtmp',
```

Comment:
Zkousel jsi to bezet v Keboola / GCP? Mozna to neni potreba a bylo to jen java specificky. Pokud tam nechame /opt/ tak ta komponenta musi mit povolenout rootUser featuru jinak to spadne